.github/workflows/molecule-test.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: "Molecule Test"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches: [ main ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read # access to check out code and install dependencies
+
+jobs:
+  test:
```

Comment:
Can we call the job molecule instead of test?  And add this line underneath it so that it uses the self-hosted runner.
runs-on: [ self-hosted,systemd-runner ]